### PR TITLE
Add style registry

### LINKED ISSUE #34: Utility class names will not update if contents change for the same class
If a `ConfigProvider` is used to `updateConfig`:
> https://github.com/cdonohue/benefit/blob/master/packages/benefit-react/ConfigContext.tsx#L33-L41

If any style declarations change for existing class names, there is a chance that two identical class names will exist (each with different declarations). And, the one inserted last will always prevail.

### LINKED ISSUE #35: SSR support
Server rendering (inline styles) should be supported in `benefit-react` out of the box and possibly configurable using `benefit` with a few additional steps.

Initially, this needs to work with
- Gatsby
- Next

### PR DESCRIPTION
**Pull request by cdonohue**
Resolves #35, Resolves #34  

This introduces a `registry` that keeps track of all `global`, `keyframes`, `utilities` (as they should be ordered), and `css` fragments.

The goal here is for benefit to add to the registry and broadcast to listeners, like inserting DOM styles. This will help to decouple actual style insertion from benefit processing.

### CODE REVIEW

File: packages/examples/with-next/pages/_app.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    let pageProps = {}
+
+    if (Component.getInitialProps) {
+      pageProps = await Component.getInitialProps(ctx)
+    }
+
+    return {
+      pageProps: { ...pageProps },
+    }
+  }
```
Comment (ericclemmons) [inline packages/examples/with-next/pages/_app.js]: This doesn't seem to be doing anything besides the defaults.  Can this be removed?

Comment (ericclemmons) [reply packages/examples/with-next/pages/_app.js]: https://github.com/zeit/next.js#custom-app

Comment (cdonohue) [reply packages/examples/with-next/pages/_app.js]: You’re absolutely right @ericclemmons.

**Comment by cdonohue** [comment]
This will also resolve #34

**Pull request merged**